Task 2/Task 2.1.2/Task 2.1.2/Program.cs

```diff
@@ -0,0 +1,804 @@
+            StartpointY = pointY;
+            Radius = radius;
+            InnerRadius = innerRadius;
+        }
+
+        private float _innerRadius;
+        public float InnerRadius 
+        {
+            get => _innerRadius;
+            set => _innerRadius = value;
+        }
+
+        public override double GetSquare
+        {
+            get
+            {
+                return Math.PI * (Radius * Radius - InnerRadius * InnerRadius);
+            }
+        }
+
+        public override double GetPerimeter
+        {
+            get
+            {
+                return 2 * Math.PI * (Radius + InnerRadius);
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"Кольцо.\n" +
+                   $"Координаты центра кольца - {StartpointX} и {StartpointY},\n" +
+                   $"внешний радиус кольца - {Radius},\n" +
+                   $"внутренний радиус кольца - {InnerRadius},\n" +
+                   $"площадь кольца - {Math.Round(GetSquare, 2, MidpointRounding.AwayFromZero)},\n" +
+                   $"суммарная длина внешней и внутренней окружностей - {Math.Round(GetPerimeter, 2, MidpointRounding.AwayFromZero)}";
+        }
+    }
+
+    class Line : OpenFigures
+    {
+        public Line (float pointX, float pointY, float endpointX, float endpointY)
+        {
+            StartpointX = pointX;
+            StartpointY = pointY;
+            EndlinepointX = endpointX;
+            EndlinepointY = endpointY;
+        }
+        public override double Length 
+        {
+            get
+            {
+                float width;
+                float heigth;
+
+                if (EndlinepointX > StartpointX)
+                {
+                    width = EndlinepointX - StartpointX;
+                }
+                else
+                {
+                    width = StartpointX - EndlinepointX;
+                }
+
+                if (EndlinepointY > StartpointY)
+                {
+                    heigth = EndlinepointY - StartpointY;
+                }
+                else
+                {
+                    heigth = StartpointY - EndlinepointY;
+                }
+
+                return Math.Sqrt(width * width + heigth * heigth);
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"Прямая линия.\n" +
+                   $"Координаты начала линии - (x:, y:) {StartpointX} {StartpointY},\n" +
+                   $"координаты окончания линии - (x:, y:) {EndlinepointX} {EndlinepointY},\n" +
+                   $"длина линии - {Math.Round(Length, 2, MidpointRounding.AwayFromZero)}";
+        }
+    }
+
+    class Triangle : FlatFigures
+    {
+        public Triangle (float pointX, float pointY, float sideA, float sideB, float sideC)
+        {
+            StartpointX = pointX;
+            StartpointY = pointY;
+            SideA = sideA;
+            SideB = sideB;
+            SideC = sideC;
+        }
```

Comment:
I can ignore your validation methods and create a thing that is *not* a valid triangle.